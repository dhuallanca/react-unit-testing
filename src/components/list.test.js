import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import List from './list';

describe('Probando List component', () => {
  test('probar nodos', () => {
    const frutas = [
      { name: 'Fresa', id: 1 },
      { name: 'Sandilla', id: 2 },
      { name: 'Naranja', id: 3 }
    ]
    const wrapper = shallow(<List title='Frutas' list={frutas}></List>);

    expect(wrapper.find('h1').exists()).toBe(true);
    expect(wrapper.find('h1').hasClass('new')).toBe(true);
    expect(wrapper.find('ul').children().length).toBe(3);
    expect(wrapper.find('ul').childAt(1).text()).toBe('Sandilla');
  });

  test('validando actualizaciones en props', () => {
    const frutas = [
      { name: 'Fresa', id: 1 },
      { name: 'Sandilla', id: 2 },
      { name: 'Naranja', id: 3 }
    ]
    const wrapper = shallow(<List title='Frutas' list={frutas}></List>);

    expect(wrapper.find('li').length).toBe(3);

    wrapper.setProps({
      list: [
        { name: 'Melocoton', id: 1 },
      ]
    });

    expect(wrapper.find('li').length).toBe(1);

    wrapper.setProps({
      title: 'Listado de frutas'
    });
    expect(wrapper.find('.new').text()).toEqual('Listado de frutas');
  });

  test('probar nodos con snapshot', () => {
    const frutas = [
      { name: 'Fresa', id: 1 },
      { name: 'Sandilla', id: 2 },
      { name: 'Naranja', id: 3 }
    ]
    const wrapper = shallow(<List title='Frutas' list={frutas}></List>);

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('probar nodos con snapshot using toJson', () => {
    const frutas = [
      { name: 'Fresa', id: 1 },
      { name: 'Sandilla', id: 2 },
      { name: 'Naranja', id: 3 }
    ]
    const wrapper = shallow(<List title='Frutas' list={frutas}></List>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
})