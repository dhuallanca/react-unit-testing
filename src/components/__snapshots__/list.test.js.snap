// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando List component probar nodos con snapshot 1`] = `"<div><h1 class=\\"new\\">Frutas</h1><ul><li>Fresa</li><li>Sandilla</li><li>Naranja</li></ul></div>"`;

exports[`Probando List component probar nodos con snapshot using toJson 1`] = `
<div>
  <h1
    className="new"
  >
    Frutas
  </h1>
  <ul>
    <li
      key="1"
    >
      Fresa
    </li>
    <li
      key="2"
    >
      Sandilla
    </li>
    <li
      key="3"
    >
      Naranja
    </li>
  </ul>
</div>
`;
